PackageIndexer/DotnetPackageIndex.cs

```diff
@@ -119,36 +120,48 @@ private static async Task<IReadOnlyList<PackageIdentity>> GetPackagesFromOtherGa
         return identities.ToArray();
     }
 
-    private static IReadOnlyList<PackageIdentity> GetLatestVersions(IReadOnlyList<PackageIdentity> identities, bool usePreviewVersions)
+    private static IReadOnlyList<PackageIdentity> GetLatestVersions(
+        IReadOnlyList<(PackageIdentity, bool)> identities,
+        bool usePreviewVersions
+        )
     {
         var result = new List<PackageIdentity>();
 
-        IEnumerable<IGrouping<string, PackageIdentity>> groups = identities.GroupBy(i => i.Id);
+        IEnumerable<IGrouping<string, (PackageIdentity, bool)>> groups = identities.GroupBy(i => i.Item1.Id).OrderBy(g => g.Key);
 
-        foreach (IGrouping<string, PackageIdentity> group in groups.OrderBy(g => g.Key))
+        foreach (IGrouping<string, (PackageIdentity, bool)> group in groups)
```

Comment:
I can't tell for certain, but the logic in this loop might be more clear using pattern matching. It looks like this could be a function of deprecation, using review packages, and the existence of a newer package. That might be a pattern match. Then, if the result isn't null, add it to the result.

I'm not sure though. Sometimes I've tried pattern matching and it's much more clear. Other times, it's kind of a mess.

Reply:
I'm going to leave this as-is for now.